src/zarr/group.py

```diff
@@ -41,6 +39,26 @@ def parse_attributes(data: Any) -> dict[str, Any]:
     raise TypeError(msg)
 
 
+@overload
+def _parse_async_node(node: AsyncArray) -> Array: ...
+
+
+@overload
+def _parse_async_node(node: AsyncGroup) -> Group: ...
+
+
+def _parse_async_node(node: AsyncArray | AsyncGroup) -> Array | Group:
+    """
+    Wrap an AsyncArray in an Array, or an AsyncGroup in a Group.
+    """
+    if isinstance(node, AsyncArray):
+        return Array(node)
+    elif isinstance(node, AsyncGroup):
+        return Group(node)
+    else:
+        assert False
```

Comment:
we could consider adding a `to_sync()` method to `AsyncGroup` and `AsyncArray` to avoid the instance checks.

Reply:
yeah, that could be convenient. I will keep this in mind if these `isinstance` checks start causing undue friction